cbmc/proofs/IotMqtt_Init/IotMqtt_Init_harness.c

```diff
@@ -0,0 +1,11 @@
+#include "iot_config.h"
+#include "private/iot_mqtt_internal.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+void harness()
+{
+  IotMqttError_t status = IotMqtt_Init();
+  assert(status == IOT_MQTT_SUCCESS || status == IOT_MQTT_NOT_INITIALIZED);
```

Comment:
Don't you have an `is_valid_mqtt` or anything you can assert here?

Reply:
I can't think of another place in the project where we would use a predicate like this.

Reply:
@danielsn Basically, `IotMqtt_Init`sets `_initCalled != MQTT_LIBRARY_UNINITIALIZED`. Thus, we could check here the following:
```c
assert(_checkInit()); // checks if _initCalled != MQTT_LIBRARY_UNINITIALIZED
```
However, the problem is that the `_initCalled` is volatile, so CBMC will *always* treat it as non-deterministic. See,
```c
static volatile uint32_t _initCalled = MQTT_LIBRARY_UNINITIALIZED;
```
I'd keep the proof harness as it is until we have another approach to treat volatile variables on CBMC.

Reply:
I think this is fine.